lib/travis/build/appliances/vault_connect.rb

```diff
@@ -0,0 +1,27 @@
+require 'travis/build/appliances/base'
+require 'travis/services/vault'
+
+module Travis
+  module Build
+    module Appliances
+      class VaultConnect < Base
+        ERROR_MESSAGE = ["Failed to connect to the Vault instance. Please verify if:\n* The Vault Token is correct. \n* The Vault token is not expired. \n* The Vault can accept connections from the Travis CI build job environments (https://docs.travis-ci.com/user/ip-addresses/).", ansi: :red]
+        SUCCESS_MESSAGE = ['Connected to Vault instance.', ansi: :green]
+
+        def apply?
+          @vault = config[:vault]
+        end
+
+        def apply
+          ENV['VAULT_ADDR'] = @vault[:api_url]
```

Comment:
Is that you had on mind when you told me abbout setting VAULT_ADDR and VAULT_TOKEN variables or I just invoke `sh.export` here?